packages/@aws-cdk/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -0,0 +1,165 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+
+/**
+ * A Lambda-based custom resource handler that provisions S3 bucket
+ * notifications for a bucket.
+ *
+ * The resource property schema is:
+ *
+ * {
+ *   BucketName: string, NotificationConfiguration: { see
+ *   PutBucketNotificationConfiguration }
+ * }
+ *
+ * For 'Delete' operations, we send an empty NotificationConfiguration as
+ * required. We propagate errors and results as-is.
+ *
+ * Sadly, we can't use @aws-cdk/aws-lambda as it will introduce a dependency
```

Comment:
That's terrible, and a problem with this pattern in general. Feels like we should extract to a separate package then.

Reply:
I wouldn't say it's a general problem. I don't expect many custom resources will be needed in these layers (at least I hope so).